python/servo/build_commands.py

```diff
@@ -238,6 +240,38 @@ def build(self, target=None, release=False, dev=False, jobs=None,
             env['OPENSSL_LIB_DIR'] = openssl_dir
             env['OPENSSL_INCLUDE_DIR'] = path.join(openssl_dir, "include")
             env['OPENSSL_STATIC'] = 'TRUE'
+            # Android builds also require having the gcc bits on the PATH and various INCLUDE
+            # path munging if you do not want to install a standalone NDK. See:
+            # https://dxr.mozilla.org/mozilla-central/source/build/autoconf/android.m4?q=android.m4&redirect_type=direct#139-161
+            os_type = platform.system().lower()
+            cpu_type = platform.machine().lower()
+            if os_type == "linux":
+                if cpu_type in ["i386", "i486", "i686", "i768", "x86"]:
+                    host = "linux-x86"
+                elif cpu_type in ["x86_64", "x86-64", "x64", "amd64"]:
+                    host = "linux-x86_64"
+                else:
+                    host = "linux-unknown"
+            elif os_type == "darwin":
+                if cpu_type in ["i386", "i486", "i686", "i768", "x86"]:
+                    host = "darwin-x86"
+                elif cpu_type in ["x86_64", "x86-64", "x64", "amd64"]:
+                    host = "darwin-x86_64"
+                else:
+                    host = "darwin-unknown"
+            else:
+                raise Exception("Android cross builds are only supported on Linux and macOS.")
+            env['PATH'] = path.join(env['ANDROID_NDK'], "toolchains", "arm-linux-androideabi-4.9",
```

Comment:
This string formatting is quite complex, I'd prefer using `.format()` like so:
```
"{andorid_path}:{path}".format(android_path=path.join(...), path=env['PATH'])
```

But I guess it's OK to leave it as-is.